### PR TITLE
[MB-5920] Redux Refactor: Moves & MTOShipments (read)

### PR DESCRIPTION
**Pull request by suzubara**
## Description

This PR updates all code that selects a logged in service member's current move and current orders to select from `state.entities.moves, state.entities.orders` and not from the legacy reducers (`state.moves, state.orders`). This ended up being a bit larger than expected, due to some dependencies between moves and orders data. Changes include:
- Move selector functions into `store/entities/selectors` and renamed to be consistent:
    - `selectActiveOrLatestOrders, selectActiveOrLatestOrdersFromEntities >> selectCurrentOrders`
    - `selectActiveOrLatestMove >> selectCurrentMove`
    - `moveIsApproved` >> `selectMoveIsApproved`
    - `lastMoveIsCanceled` >> `selectHasCanceledMove`
    - `selectedMoveType` >> `selectMoveType`
    - `selectMTOShipmentsByMoveId` >> `selectMTOShipmentsForCurrentMove`
- Added an `onboarding` reducer in order to save the `conusStatus` value. This was because this value is not saved by the API at all, and should be stored separately from cached API data in Redux. I'm not sure that we need to save this value in Redux at all, but for now moving it into a different reducer felt like the path with the least friction.
    - Also migrated the selector: `selectedConusStatus` >> `selectConusStatus`
- I also needed to update the submit handler on the `Orders` step of onboarding, because it was navigating to the next page (Upload Orders) before the create request was completed. I took this opportunity to migrate the submit handler to use sagas (same as the other steps of onboarding now), and to update the service member if creating orders.
- Deprecated `selectUploadsForActiveOrders` - I don't think this selector was working as-is, resulting in existing uploads not showing up in the customer & office apps. It was looking for documents & uploads in entities, neither of which get populated as far as I can tell. I switched to alternative selectors that select the uploads nested within the orders object.
- Added some more customer prop types shapes
- Deleted legacy code that is no longer referenced anywhere
- Miscellaneous eslint fixes on touched files

## Reviewer Notes

After this PR, there should be no references made to Redux `state.moves` or `state.orders`. There are a few lingering methods used in the legacy ducks files, which can be migrated and/or deleted in the follow up stories to delete old reducers. To test, you can go through any service member flows and especially test that the add orders and add shipments work correctly. One note is that there are existing bugs related to _editing_ existing orders, so that is not expected to be fixed in this PR. There should be no new bugs introduced, and no visible changes to the user.

## Setup

To start the application:

```
make server_run
make client_run
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5920) for this change

### CODE REVIEW

**Comment by suzubara** [comment]
re: the dangerJS lint warning, I think is a false positive. the only eslint disables I added were `camelcase` & `import/prefer-default-export` which are on the allowed list

File: src/components/Customer/Review/Summary/index.jsx [hunk 1]
```diff
@@ -32,8 +25,16 @@ import SectionWrapper from 'components/Customer/SectionWrapper';
 import NTSShipmentCard from 'components/Customer/Review/ShipmentCard/NTSShipmentCard';
 import NTSRShipmentCard from 'components/Customer/Review/ShipmentCard/NTSRShipmentCard';
 import { showLoggedInUser as showLoggedInUserAction } from 'shared/Entities/modules/user';
-import { selectMTOShipmentsByMoveId } from 'shared/Entities/modules/mtoShipments';
-import { selectServiceMemberFromLoggedInUser } from 'store/entities/selectors';
+import {
+  selectServiceMemberFromLoggedInUser,
```
Comment (Ronolibert) [inline src/components/Customer/Review/Summary/index.jsx:29]: Love the standardization of the naming convention!

**Review by Ronolibert — CHANGES_REQUESTED** [review]
Had a more philosophical question I guess. I notice everything has normalize set to false, what situations would we and would we not want to normalize the data?

**Review by Ronolibert — APPROVED** [review]
Makes sense! Everything works the same and changes look good! So much change of old cold, so cathartic. Really like how everything is organized

**Review by lynzt — APPROVED** [review]
LGTM.. based on the pr walk thru, i'm good. 
I didn't have time to run it but I see you got a 👍 from Ron, so 🚢

**Pull request merged**